pkg/teeth/github.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teeth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/abcxyz/github-metrics-aggregator/pkg/githubclient"
+	"github.com/google/go-github/v56/github"
+	"github.com/sethvargo/go-retry"
+)
+
+const ()
+
+var (
+	// can be overriden for testing
+	retryMinWaitDuration        = 1 * time.Second
```

Comment:
I recall GitHub having org-wide rate limits for comment API, so just be careful when metering requests that you take into account retries

Reply:
Ack see my other comment about RateLimitErrors. Will resolve there.